### PR TITLE
Remove dependencies and move from dapp to forge

### PR DESCRIPTION
**Pull request by gbalabasquer**

### CODE REVIEW

**Comment by gbalabasquer** [comment]
Once this gets merged I'll move on with a new PR for upgrading to solc 0.8.x.

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-build        :; ./build.sh
+build        :; DAPP_BUILD_OPTIMIZE=1 DAPP_BUILD_OPTIMIZE_RUNS=200 dapp --use solc:0.6.12 build
```
Comment (iamchrissmith) [inline Makefile:1]: if we are moving from dapp to forge, do we want to switch these too:
```
forge build
forge clean
forge create
```
etc?

**Review by iamchrissmith — CHANGES_REQUESTED** [review]
one clean up item and two questions

:mechanical_arm:

File: src/test/DssAction.t.sol [hunk 2]
```diff
@@ -195,9 +505,9 @@ contract ActionTest is DSTest {
     function dai(address urn) internal view returns (uint) {
         return vat.dai(urn) / RAY;
     }
-    function gem(bytes32 ilk, address urn) internal view returns (uint) {
-        return vat.gem(ilk, urn);
-    }
+    // function gem(bytes32 ilk, address urn) internal view returns (uint) {
+    //     return vat.gem(ilk, urn);
+    // }
```
[the remark this answers is not in the document]
Comment (gbalabasquer) [reply src/test/DssAction.t.sol]: Done!

File: src/test/DssAction.t.sol [hunk 3]
```diff
@@ -20,138 +20,448 @@
...
+interface Univ2OracleFactoryLike {
+    function build(address, address, bytes32, address, address) external returns (address oracle);
+}
+
+interface D3MLike {
+    function bar() external view returns (uint256);
+    function rely(address) external;
+}
+
+contract MockUniPair {
```
[the remark this answers is not in the document]
Comment (gbalabasquer) [reply src/test/DssAction.t.sol]: Done!

File: Makefile (hunk 1 above)
Comment (gbalabasquer) [reply Makefile:1]: the thing is `forge create` apparently doesn't work the same than `dapp create` (we are still using the last one I believe). And we need to `build` using `dapp` in order to be able to create with it.
I'd guess `dapp clean` could be replaced without any problem.

Comment (gbalabasquer) [reply Makefile:1]: Yeah actually, `forge clean` also removed `cache` directory which should be fine. For now I'm changing only that one.

Comment (gbalabasquer) [reply Makefile:1]: Done!

**Review by iamchrissmith — APPROVED** [review]
lgtm!

**Review by rockyfour — APPROVED** [review]
Changes LGTM.
Tests pass.

File: src/test/Mocks.sol [hunk 4]
```diff
@@ -0,0 +1,366 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
```
Comment (naszam) [inline src/test/Mocks.sol:1]:
```suggestion
// SPDX-FileCopyrightText: © 2022 Dai Foundation <www.daifoundation.org>
// SPDX-License-Identifier: AGPL-3.0-or-later
```

File: src/test/Mocks.sol [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+//
+// Mocks.sol -- DSS Executive Spell Action Tests
+//
+// Copyright (C) 2022 Dai Foundation
```
Comment (naszam) [inline src/test/Mocks.sol:5]:
```suggestion
```

**Pull request merged**